spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-stream-functional-binder-sample/source/src/main/java/com/example/Source.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.function.Supplier;
+
+import com.example.model.UserMessage;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+
+/**
+ *
+ */
+@SpringBootApplication
+public class Source {
+
+	@Autowired
+	@Qualifier("postOffice")
+	private ConcurrentLinkedQueue<UserMessage> postOffice;
+
+	@Bean
+	Supplier<UserMessage> generateUserMessages() {
+		return () -> postOffice.poll();
```

Comment:
@olegz Actually, now that I think about it more, is "inputting data from UI and sending it into a stream" a valid Spring Cloud Stream usage scenario at all?
In real world, it would be something like files arriving into file system, which you'd hook up with a SCS `Processor`, so it would not be a `Source` at all.

Reply:
Yes, very valid and very possible. Here is all the code you need
```
@SpringBootApplication
@Controller
@RequestMapping("/home")
public class DemoFunctionStreamWebApplication {

	public static void main(String[] args) {
		SpringApplication.run(DemoFunctionStreamWebApplication.class, args);
	}

	EmitterProcessor<String> processor = EmitterProcessor.create();

	@RequestMapping(value="/send")
	@ResponseBody
	public String foo(@RequestParam("name") String name){
		System.out.println("Hello " + name);
		processor.onNext(name);
		return name;
	}

	@Bean
	public Supplier<Flux<String>> supplier() {
		return () -> processor;
	}
}
```
Is that what you were asking about?

Reply:
Yes! I'll use this way instead of opening a socket connection; this is simpler.
I like `EmitterProcessor` better than the queue, too, so I'll use it in the example. I did not think to use reactor primitives.